component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/chain/ChainedInput.java

```diff
@@ -30,11 +33,7 @@ public final class ChainedInput implements Input {
     public Object next() {
         while (true) {
             if (delegate == null) {
-                delegate = parent.getIterator().hasNext() ? parent.getIterator().next().create() : null;
```

Comment:
What if the end of while(true) where it set delegate to null is executed? I think a previous logic expected that we will get a next Input in chain, but now we don't have that logic. So we will read only from the first Input in chain, and just ignore the rest.
Did I miss smth?

Reply:
delegate should be not null if lifecycle is respected. This case should not happened.